modules/standard/CTypes.chpl

```diff
@@ -703,7 +728,8 @@ module CTypes {
   }
   @chpldoc.nodoc
   inline operator c_ptrConst.!=(a: c_ptrConst, b: c_ptrConst)
-      where a.eltType == b.eltType || a.eltType == void || b.eltType == void {
+      where a.type != chpl_c_string && b.type != chpl_c_string &&
```

Comment:
ditto

Reply:
following advice in https://github.com/chapel-lang/chapel/pull/22622#discussion_r1283709792 made this addition to the `where` clause unnecessary